services/spar/src/Spar/API.hs

```diff
@@ -165,12 +100,36 @@ authreqPrecheck msucc merr idpid = validateAuthreqParams msucc merr
                                 *> SAML.getIdPConfig idpid
                                 *> return NoContent
 
-authreq :: NominalDiffTime -> Maybe URI.URI -> Maybe URI.URI -> SAML.IdPId -> Spar (SAML.FormRedirect SAML.AuthnRequest)
-authreq authreqttl msucc merr idpid = do
+authreq :: NominalDiffTime -> DoInitiate -> Maybe UserId -> Maybe URI.URI -> Maybe URI.URI -> SAML.IdPId
+        -> Spar (WithBindCookie (SAML.FormRedirect SAML.AuthnRequest))
+authreq _ DoInitiateLogin (Just _) _ _ _ = throwSpar SparInitLoginWithAuth
+authreq _ DoInitiateBind Nothing _ _ _   = throwSpar SparInitBindWithoutAuth
+authreq authreqttl _ zusr msucc merr idpid = do
   vformat <- validateAuthreqParams msucc merr
-  form@(SAML.FormRedirect _ ((^. SAML.rqID) -> reqid)) <- SAML.authreq authreqttl sparRequestIssuer idpid
+  form@(SAML.FormRedirect _ ((^. SAML.rqID) -> reqid)) <- SAML.authreq authreqttl sparSPIssuer idpid
   wrapMonadClient $ Data.storeVerdictFormat authreqttl reqid vformat
-  pure form
+  cky <- initializeBindCookie zusr authreqttl
+  SAML.logger SAML.Debug $ "setting bind cookie: " <> show cky
+  pure $ addHeader cky form
+
+-- | Create bind cookie with 60 minutes life expectancy; *iff* the user is already authenticated,
+-- store it with the bind cookies.  If user already has a 'UserSSOId' (need to ask brig for that),
+-- throw an error.
+initializeBindCookie :: Maybe UserId -> NominalDiffTime -> Spar BindCookie
+initializeBindCookie zusr authreqttl = do
```

Comment:
I think we should initialise/set the path/domain during spar's startup; these values never change during runtime and this way you will see config errors only when users make requests which is a bit too late